js/geo/transform.js

```diff
@@ -408,6 +408,9 @@ class Transform {
         mat4.perspective(m, 2 * Math.atan((this.height / 2) / this.altitude), this.width / this.height, 0.1, farZ);
         mat4.translate(m, m, [0, 0, -this.altitude]);
 
+        // a hack around https://github.com/mapbox/mapbox-gl-js/issues/2270
+        m[14] = Math.min(m[14], m[15]);
```

Comment:
There seem to be no intuitive meaning since the perspective projection math is [a bit involved](http://www.songho.ca/opengl/gl_projectionmatrix.html), but at this point of position matrix calculation, `m[15]` always equals the altitude value (`1.5`), and `m[14]` represents translation along Z axis (which always seems to be `<=1.5`).